main.go

```diff
@@ -14,128 +10,235 @@ const GLOBAL_NAME = "sqlparser"
-				go func() interface{} {
-					if len(statement) > maxQuerySize {
-						return reject.Invoke(Error.New("statement size larger than specified max"))
-					}
-					ast, err := sqlparser.Parse(statement)
-					if err != nil {
-						return reject.Invoke(Error.New(err.Error()))
-					}
-					if len(ast.Statements) == 0 {
-						return reject.Invoke(Error.New("the statement is empty"))
-					}
-					response := make(map[string]interface{})
-					statements := make([]interface{}, len(ast.Statements))
-					// Since we support write queries with more than one statement,
-					// check each of them, and for write queries, check
-					// that all statements reference the same table.
-					var targetKind, refKind, targetTable, refTable string
-					for i, stmt := range ast.Statements {
-						if ast.Errors[i] != nil {
-							return reject.Invoke(Error.New("non sysntax error encountered: " + ast.Errors[i].Error()))
-						}
-						switch s := stmt.(type) {
-						case sqlparser.CreateTableStatement:
-							node := s.(*sqlparser.CreateTable)
-							refTable = node.Table.String()
-							refKind = "create"
-						case sqlparser.ReadStatement:
-							refKind = "read"
-						case sqlparser.GrantOrRevokeStatement:
-							refTable = s.GetTable().String()
-							refKind = "acl"
-						case sqlparser.WriteStatement:
-							refTable = s.GetTable().String()
-							refKind = "write"
-						default:
-							reject.Invoke(Error.New("unknown statement type"))
-						}
-						if targetKind == "" {
-							targetKind = refKind
-						} else if targetKind != refKind {
-							targetKind = "write"
-						}
-						if targetTable == "" {
-							targetTable = refTable
-						} else if targetTable != refTable {
-							err := Error.New("queries are referencing two distinct tables: " + targetTable + " " + refTable)
-							reject.Invoke(err)
-						}
-						statements[i] = stmt.String()
+			if !nameMap.IsUndefined() {
+				_ = UpdateTableNames(ast, func(name string) (string, bool) {
+					value := nameMap.Get(name)
+					if value.IsUndefined() {
+						return "", false
 					}
-					if targetTable != "" {
-						response["table"] = targetTable
-					}
-					response["statements"] = statements
-					response["type"] = targetKind
-					return resolve.Invoke(js.ValueOf(response))
-				}()
-				return nil
-			})
-			return js.Global().Get("Promise").New(handler)
-		}),
-		"maxQuerySize": js.FuncOf(func(this js.Value, args []js.Value) interface{} {
-			if len(args) < 1 {
-				return maxQuerySize
+					return value.String(), true
+				})
 			}
-			maxQuerySize = args[0].Int()
-			return maxQuerySize
-		}),
-		"structureHash": js.FuncOf(func(this js.Value, args []js.Value) interface{} {
-			if len(args) < 1 {
-				return js.Global().Get("Promise").Call("reject", Error.New("missing required argument 'statement'"))
+			updated := ast.String()
+			// Re-parse the statement in case we invalidated the names
+			if _, err = sqlparser.Parse(updated); err != nil {
```

Comment:
Since we already know that the statement is syntactically valid because of the first parsing pass, you could just call `[ValidateTargetTables(ast)](https://github.com/tablelandnetwork/go-sqlparser/blob/main/helpers.go#L61)`. That would validate all tables in that statement and return a slice of valid tables, that you can use to extract the chain id and do other checks. Makes sense?

Reply:
Yes it does. Also note that I ended up putting this back into WIP because, this approach isn't going to work all that well if multiple passes over the sql statement are used.

Reply:
yeh that's what I'm doing now 👍